### ISSUE TITLE
Python3 ImportError

### ISSUE THREAD

**Issue by vccolombo** [Zarad1993/dyc#8]
Using dyc with Python3 gives the following error:

> ImportError: cannot import name 'ParsedConfig' from 'parser'

As Python2 is about to be deprecated ([https://pythonclock.org/](url)) and more people are using Python 3, I think it'll be great to fix this issue

I took a look at the code and tested some things, and I came with a solution: change all imports from absolute to relative, as stated here [https://www.python.org/dev/peps/pep-0328/#relative-imports-and-indirection-entries-in-sys-modules](url). It would require changing the imports in the following files:

- methods.py
- base.py
- diff.py
- dyc.py
- events.py
- main.py
- methods.py

If you think there is no problem doing this, can I open the PR?
Thank you very much

**Comment by mboekhold** [comment]
I'm not 100% certain that this is an issue that is unintentional, in the readme @Zarad1993 mentions that it specifically targets python2.7. I do agree that it should be target for more recent python versions for the reasons you mentioned

**Comment by vccolombo** [comment]
Thanks for you reply!
Indeed. Still, I think this change would keep python2 working as intended if the project keeps targeting only python2, but would make the program compatible with python3 as well, which would be great 
Still, I thought a commit described with "try to fix pip" was actually an attempt to fix this issue, but I may have misunderstood it

**Comment by Zarad1993** [comment]
@vccolombo - Thank you so much for bringing this up. It is definitely wonderful to have the project compatible with both versions! As @mboekhold said, it was targeting 2.7 from the start. But at some point it should also support Python 3, more importantly at early stages before the codebase grows larger.

I'd be happy to review/merge your PR once ready.
Thanks!

**Comment by vccolombo** [comment]
Hello again!

After implementing the changes I proposed, I tested it more deeply and encountered many other problems that give errors in Python 3. For example:

Line 23 of base.py `found = len(filter(lambda word: word.lstrip() in keywords, line.split(' '))) > 0`. filter does not return a list in Python 3 as it does in Python 2. This means that len(filter) returns an error. This very same issue happens in line 98 of utils.py (I don't get exactly why), line 228 of methods.py and others as well. 
May I extend the work trying to fix these issues to make it compatible with both Python 3 and 2?
Thanks!
